src/App.js

```diff
@@ -1,13 +1,53 @@
 import React, { Component } from 'react'
 import { Instructions } from './components'
+import { Counter } from './components'
+import { InitialCountForm } from './components'
 import './styles/app.css'
 
 class App extends Component {
+  constructor(props) {
+    super(props)
+    this.onInputChange = this.onInputChange.bind(this)
```

Comment:
you can autobind this (i.e. you don't have to write this line) if you write your function signature like this instead:

```javascript
onInputChange = e => {
   ...
```